### PR TITLE
Disable pre dex and reduce parallelism for CI builds

### PR DESCRIPTION
**Pull request by twyatt**
Attempting to further reduce memory usage during CircleCI builds to address [lingering issues](https://github.com/ponewheel/android-ponewheel/issues/28#issuecomment-351344075).

Disables pre dex which is [unnecessary for CI builds]:

> By default the Gradle android plugin pre-dexes dependencies, converting their Java bytecode into Android bytecode. This speeds up development greatly since gradle only needs to do incremental dexing as you change code.
>
> Because CircleCI always runs clean builds this pre-dexing has no benefit; in fact it makes compilation slower and can also use large quantities of memory. We recommend disabling pre-dexing for Android builds on CircleCI.

See also: [Building multi-dex Android project gets OOM](https://discuss.circleci.com/t/building-multi-dex-android-project-gets-oom/315)

[unnecessary for CI builds]: https://circleci.com/docs/1.0/android/#disable-pre-dexing-to-improve-build-performance

### CODE REVIEW

**Comment by ebabel** [comment]
Unfortunately this build still has the same error.

**Comment by twyatt** [comment]
Found that the likely cause of this issue is a [bug in the Android gradle plugin](https://issuetracker.google.com/issues/69380617).

I don't want to assume that there aren't potential implications of updating the Android gradle plugin version, so feel free the bump the version if you want, but the bug sounds to be minor (and can be ignored if you're not ready to upgrade your plugin version).

TL;DR bug in Android gradle plugin versions prior to `3.1.0-alpha5` erroneously emit a stacktrace when project targets `1.8`

**Comment by ebabel** [comment]
> when project targets 1.8

1.8?

**Comment by ebabel** [comment]
Looks like the latest is [3.1.0-alpha5](https://dl.google.com/dl/android/maven2/index.html). Maybe we'll just wait for a beta or rc and ignore for now.

**Comment by twyatt** [comment]
By `1.8` I meant:

```
targetCompatibility JavaVersion.VERSION_1_8
```
